uitools/cpp/Esri/ArcGISRuntime/Toolkit/Internal/DoOnLoad.h

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ *  Copyright 2012-2020 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#ifndef ESRI_ARCGISRUNTIME_TOOLKIT_INTERNAL_DOONLOAD_H
+#define ESRI_ARCGISRUNTIME_TOOLKIT_INTERNAL_DOONLOAD_H
+
+// ArcGISRuntime headers
+#include <Loadable.h>
+
+// Qt headers
+#include <QObject>
+
+// C++ headers
+#include <type_traits>
+
+namespace Esri {
+namespace ArcGISRuntime {
+namespace Toolkit {
+
+  /*
+    \internal
+    \brief Executes method \a f immediately if \a sender is loaded, otherwise
+    attempts to load \a sender and then executes \a f after the load occurs.
+    Returns a connection object which may be default-constructed.
+
+    Example:
+    \begincode
+    doOnLoad(floorManager, this, []
```

Comment:
maybe this should be `doOnLoaded` since the action takes place when load completes (e.g. status is loaded) and not when you call `load()`?

Reply:
this is a good comment! As this is used across a couple of these PRs I'll make this change upstream.